packages/js/src/notifications/helpers.ts

```diff
@@ -261,3 +262,107 @@ const getNotificationDetails = (
     };
   }
 };
+
+export const readAll = async ({
+  emitter,
+  inboxService,
+  notificationsCache,
+  tags,
+}: {
+  emitter: NovuEventEmitter;
+  inboxService: InboxService;
+  notificationsCache: NotificationsCache;
+  tags?: NotificationFilter['tags'];
+}): Result<void> => {
+  try {
+    const notifications = notificationsCache.getUniqueNotifications({ tags });
+    const optimisticNotifications = notifications.map(
+      (notification) =>
+        new Notification({
+          ...notification,
+          isRead: true,
+          readAt: new Date().toISOString(),
+          isArchived: false,
+          archivedAt: undefined,
+        })
+    );
+    emitter.emit('notifications.read_all.pending', { args: { tags }, data: optimisticNotifications });
+
+    await inboxService.readAll({ tags });
+
+    emitter.emit('notifications.read_all.resolved', { args: { tags }, data: optimisticNotifications });
+
+    return {};
+  } catch (error) {
+    emitter.emit('notifications.read_all.resolved', { args: { tags }, error });
+
+    return { error: new NovuError('Failed to read all notifications', error) };
+  }
+};
+
+export const archiveAll = async ({
+  emitter,
+  inboxService,
+  notificationsCache,
+  tags,
+}: {
+  emitter: NovuEventEmitter;
+  inboxService: InboxService;
+  notificationsCache: NotificationsCache;
+  tags?: NotificationFilter['tags'];
+}): Result<void> => {
+  try {
+    const notifications = notificationsCache.getUniqueNotifications({ tags });
+    const optimisticNotifications = notifications.map(
+      (notification) =>
+        new Notification({
+          ...notification,
+          isRead: true,
+          readAt: new Date().toISOString(),
+          isArchived: true,
+          archivedAt: new Date().toISOString(),
+        })
+    );
+    emitter.emit('notifications.archive_all.pending', { args: { tags }, data: optimisticNotifications });
+
+    await inboxService.archiveAll({ tags });
+
+    emitter.emit('notifications.archive_all.resolved', { args: { tags }, data: optimisticNotifications });
+
+    return {};
+  } catch (error) {
+    emitter.emit('notifications.archive_all.resolved', { args: { tags }, error });
+
+    return { error: new NovuError('Failed to archive all notifications', error) };
+  }
+};
+
+export const archiveAllRead = async ({
```

Comment:
❓  Why is this function needed? Couldn't we pass a filter to `archiveAll()` and avoid the extra functions ?

Reply:
these helper functions create different optimistically updated objects, send different events, and call different api... too many if statements